ui.js

```diff
@@ -0,0 +1,67 @@
+const express = require('express')  
+var bodyParser = require('body-parser');
+var request = require('request').defaults({ encoding: null });
+var jade = require('jade');
+var fs = require('fs');
+var fileUpload = require('express-fileupload')
+
+const app = express()  
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(fileUpload())
+app.set('view engine', 'jade')
+
+
+const port = process.env.MICRO_UI_PORT || process.env.PORT || 80;
+const masterEndpoint = process.env.MICRO_MASTER_ENDPOINT || "http://micro-master.default.svc.cluster.local";
+const uploadImageUrl = masterEndpoint + '/task'
+
+function startService() {
+	app.listen(port, (err) => {
+		if (err) {
+			return console.log('something bad happened', err)
+		}
+		console.log(`Ui is listening on ${port}`)
+	})
+}
+
+app.get('/', (req, res) => {
+	var taskId = req.query.taskId;
+	taskId = taskId ? taskId : "0";
+	downloadImageUrl = `${masterEndpoint}/task/${taskId}/download`;
+
+	request.get({url: downloadImageUrl, timeout: 3000}, function (error, response, body) {
```

Comment:
Thing is that backend lives inside Kubernetes which allows it to access internal Services by name. This way I can download image without even knowing IP of master node by accessing master's Service by name:
```
http://micro-master.default.svc.cluster.local
```
The endpoint above is not accessible from outisde Kubernetes (where browser lives).